okta/data_source_okta_user.go

```diff
@@ -112,22 +112,14 @@ func dataSourceUserRead(ctx context.Context, d *schema.ResourceData, m interface
 		return diag.Errorf("failed to set user's properties: %v", err)
 	}
 
-	skip := false
-	if val := d.Get("skip_roles"); val != nil {
-		skip = val.(bool)
-	}
-	if !skip {
+	if d.Get("skip_roles") == false {
```

Comment:
Doesn't this need to be done as a type assertion and using OK to avoid a nil exception?

```go
if ok, sr := d.GetOk("skip_roles"); ok && sr.(bool) {
// code
}
```
or possibly adding an extra check on the type assertion.

```go
if ok, sr := d.GetOk("skip_roles"); ok {
    if skipRoles, ok := sr.(bool); ok && skipRoles {
        // code
    }
}
```

Reply:
because `skip_roles` [in the schema](https://github.com/okta/terraform-provider-okta/blob/master/okta/data_source_okta_user.go#L70) has a default value I thought this might be okay?
the default value of bool false should prevent the value from ever being nil? 

I'm still very new to golang so happy to take guidance where I can get it
if you think I should update these lines, happy to make what ever changes are required 😄